### PR TITLE
[GPU] Enable BY_CHANNEL de-quantize/re-quantize support in KV_CACHE_R…

### PR DESCRIPTION
**Pull request by Prithviraj-R**
[GPU] Enable BY_CHANNEL de-quantize/re-quantize support in KV_CACHE_ROTATE OCL kernel
 
### Description of the issue(symptom, root-cause, how it was resolved)
- The existing pa_kv_cache_rotate_ref.cl kernel handles RoPE (Rotary Position Embedding) for the key cache.
- In order to do the rotation calculation, the kernel first need to de-quantize the key cache content on the fly.
- The dequantize logic differs based on whether "by-token" or "by-channel" kv cache quantization was used in the prior kv_cache_update stage.
- This PR adds support for "by-channel" dequantization before rotation and again "by-channel" re-quantize post rotation.

#### The code and line that caused this issue (if it is not changed directly)
- intel_gpu/src/graph/impls/ocl_v2/pa_kv_cache_rotate_ref.cl

#### Reproduction step and snapshot (if applicable. Do not attach for customer model)
- NA

#### Problematic graph
- NA

#### Checklist
- [X] Is it a proper fix? (not a workaround)
- [X] Did you include test case for this fix, if necessary?
- [X] Did you review existing test that can be extended to cover this scenario? Which test did you review?  None. New tests were added to cover the support this PR adds.

### Tickets:
- *CVS-170994*

### CODE REVIEW

File: src/plugins/intel_gpu/src/graph/impls/ocl_v2/pa_kv_cache_rotate_ref.cl [hunk 1]
```diff
@@ -37,7 +37,7 @@ KERNEL(pa_kv_cache_rotate)(
     __local INPUT2_TYPE rotation_coefficients[HEAD_SIZE][PAGED_ATTENTION_BLOCK_SIZE];
 
     const bool per_token_rotation = INPUT1_FEATURE_NUM == PAGED_ATTENTION_BLOCK_SIZE;
-
+
```
Comment (mklimenk) [inline src/plugins/intel_gpu/src/graph/impls/ocl_v2/pa_kv_cache_rotate_ref.cl]: Please remove the extra tab here. Otherwise looks good to me, thanks!

Comment (Prithviraj-R) [reply src/plugins/intel_gpu/src/graph/impls/ocl_v2/pa_kv_cache_rotate_ref.cl]: Done

File: src/plugins/intel_gpu/src/graph/impls/ocl_v2/pa_kv_cache_rotate_ref.cl [hunk 2]
```diff
@@ -67,74 +67,131 @@ KERNEL(pa_kv_cache_rotate)(
...
+        const uint comp_offset_first = cache_offset_first - sglid + PAGED_ATTENTION_BLOCK_SIZE;
+        const uint comp_offset_second = cache_offset_second - sglid + PAGED_ATTENTION_BLOCK_SIZE;
+
+        UNCOMPRESSED_TYPE* comp_ptr_first = key_cache + comp_offset_first;
+        UNCOMPRESSED_TYPE comp_scale_first = comp_ptr_first[0];
+        UNCOMPRESSED_TYPE comp_zp_first = comp_ptr_first[1];
+        
+        UNCOMPRESSED_TYPE* comp_ptr_second = key_cache + comp_offset_second;
+        UNCOMPRESSED_TYPE comp_scale_second = comp_ptr_second[0];
+        UNCOMPRESSED_TYPE comp_zp_second = comp_ptr_second[1];
```
Comment (e-ddykim) [inline src/plugins/intel_gpu/src/graph/impls/ocl_v2/pa_kv_cache_rotate_ref.cl:110]: nit: can we move this part after L116 for better readability?

**Review by e-ddykim — APPROVED** [review]
LGTM

**Review by andrew-k-park — APPROVED** [review]

**Comment by Prithviraj-R** [comment]
build_jenkins
